Feature/NEX-1516/NEX-1518/Migrate TestPreview provider to qti TestRunner, a proxy for TestPreview

bziondik opened the pull request:
Related to: 
https://oat-sa.atlassian.net/browse/NEX-1516
https://oat-sa.atlassian.net/browse/NEX-1518

Migration TestPreview from taoQtiTestPreviewer TestRunner provider to qti TestRunner
Was created a proxy for TestPreview
Removed changes in provider/item were done in phase 1
Added unit tests

How to test:

- switch taoQtiTest and taoTests to branch feature/NEX-1043/test-preview-base-branch
- create test
- go to Test Preview

jsconan reviewed (changes requested):
Some minor comments regarding the code, but the overall looks good.
The feature looks to work. But not all plugins are loaded (ex: the review panel).
The version bump is missing, but I guess it is ok since this is a merge to an integration branch?

Unfortunately, unit tests are failing in cycle. Sometimes the tests fail because of the CRSF token, and sometimes the test fails because of the plugin `enhancedReadOnlyMod` which cannot be loaded.

```
Testing http://127.0.0.1:63431/taoQtiTestPreviewer/views/js/test/previewer/proxy/item/test.html
........F...............
>> itemProxy - a804b399
>> Message: The proxy must update the security token
>> Actual: "token1"
>> Expected: "4567"
>> at http://127.0.0.1:63431/taoQtiTestPreviewer/views/js/test/previewer/proxy/item/test.js:169:32

Testing http://127.0.0.1:63431/taoQtiTestPreviewer/views/js/test/previewer/proxy/test/test.html
.......F.........
>> testProxy - 1c6728d7
>> Message: The proxy must update the security token
>> Actual: "token1"
>> Expected: "4567"
>> at http://127.0.0.1:63431/taoQtiTestPreviewer/views/js/test/previewer/proxy/test/test.js:168:32
```

```
Testing http://127.0.0.1:63486/taoQtiTestPreviewer/views/js/test/previewer/plugins/content/enhancedReadOnlyMode/test.html
core/eventifier Cannot read property 'scrollHeight' of null JSHandle@error JSHandle@object
>> Chrome unable to load 'http://127.0.0.1:63181/taoQtiTestPreviewer/views/js/test/previewer/plugins/content/enhancedReadOnlyMode/test.html' URI.
>> There was an error with headless chrome
Fatal error: http://127.0.0.1:63181/taoQtiTestPreviewer/views/js/test/previewer/plugins/content/enhancedReadOnlyMode/test.html
```

The same in the browser:
<img width="937" alt="image" src="https://user-images.githubusercontent.com/1500098/91027846-e386c100-e5fc-11ea-88af-e6ffdb9c3445.png">

bziondik commented:
@jsconan I agree with you that cloning looks like a dirty hack,
for the footer, I updated solution using CSS
But for the logo, I don't know how to solve this with a template. Because we customize a logo for the client in client extensions by `CLIENT_EXTENSION/views/templates/themes/platform/default/header-logo.tpl`
I moved logo cloning to plugin to have the ability to disable it in client extension if needed in the future
Because the next step will be to move the testRunner configuration on BE

jsconan reviewed (approved):
The feature works well. Unit tests look ok now.

However, I still have a concern with the header and footer. I think we should not display them as part of the test previewer. Especially because they don't add any value, and they may confuse the user as well: none of the header action actually work.
A better approach would have been to follow the same as for the item previewer: no header nor footer, and a close button to exit the preview.

I approve the broad changes, regarding the refactoring, not the UI.

**Review Checklist**
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
